astropy/table/table.py

```diff
@@ -44,7 +44,7 @@ def descr(col):
     This returns a 3-tuple (name, type, shape) that can always be
     used in a structured array dtype definition.
     """
-    col_dtype_str = col.dtype.str if hasattr(col, 'dtype') else 'O'
+    col_dtype_str = col.dtype if hasattr(col, 'dtype') else 'O'
```

Comment:
Just a niggle, but the var name should probably be called `col_dtype` now to correspond to the actual contents.